src/models.ts

```diff
@@ -540,6 +605,100 @@ export class Sequential extends Model {
 
   // TODO(cais): Override get trainableWeights() here
 
+  /**
+   * Extract weight values of the model.
+   *
+   * @param config: An instance of `io.SaveConfig`, which specifies model-saving
+   *   options such as whether only trainable weights are to be saved.
+   * @returns A `NamedTensorMap` mapping original weight names (i.e.,
+   *   non-uniqueified weight names) to their values.
+   */
+  protected getNamedWeights(config?: io.SaveConfig): NamedTensorMap {
+    const namedWeights: NamedTensorMap = {};
+
+    const trainableOnly = config != null && config.trainableOnly;
+    const weights = trainableOnly ? this.trainableWeights : this.weights;
+    const weightValues = this.getWeights(trainableOnly);
+    for (let i = 0; i < weights.length; ++i) {
+      if (trainableOnly && !weights[i].trainable) {
+        // Optionally skip non-trainable weights.
+        continue;
+      }
+      namedWeights[weights[i].originalName] = weightValues[i];
+    }
+    return namedWeights;
+  }
+
+  /**
+   * Save the model to an IOHandler.
+   *
+   * An `IOHandler` is an object that has a `save` method of the proper
+   * signature defined. The `save` method manages the storing or transmission of
+   * serialized data ("artifacts") of the model's topology and weights on or via
+   * a specific medium, such as browser downloads, local storage, IndexedDB and
+   * HTTP requests to a server. TensorFlow.js provides a number of frequently
+   * used saving mediums, such as `tf.io.browserDownloads` and
+   * `tf.io.browserLocalStorage`. See `tf.io` for more details.
+   *
+   * This method allows you to refer to certain types of `IOHandler`s as
+   * URL-like string shortcuts. See the example below for how to save a simple
+   * model to browser local storage and load it back.
+   *
+   * ```js
+   * // Define and train a model, which will be saved and then loaded later.
+   * const model = tf.sequential();
+   * model.add(tf.layers.dense({units: 1, inputShape: [1]}));
+   * model.compile({loss: 'meanSquaredError', optimizer: 'sgd'});
+   *
+   * const xs = tf.tensor2d([1, 2, 3, 4], [4, 1]);
+   * const ys = tf.tensor2d([1, 3, 5, 7], [4, 1]);
+   * await model.fit(xs, ys, {epochs: 100});
+   * model.predict(tf.tensor2d([[5]])).print();
+   *
+   * // Save the model (along with its weights) to the browser local storage,
+   * // under the unique path name 'SavedByModelSaveCodeSnippet';
+   * await model.save('localstorage://SavedByModelSaveCodeSnippet');
+   * console.log('Done saving model.');
+   *
+   * // Load the model back and use it to do prediction, which should yield
+   * // the same result as the prediction result above.
+   * model2 = await tf.loadModel('localstorage://SavedByModelSaveCodeSnippet');
```

Comment:
This shows the scheme I was expecting, but I don't think the string load side handles it yet?